app/controllers/int_api/marketplaces_controller.rb

```diff
@@ -51,7 +51,7 @@ def create
 
     #include port separately to url_params if it was present
     if marketplace[:url] && marketplace[:url].split(':')[2]
-      url_params.merge!(port: marketplace[:url].split(':')[2])
+      url_params[:port] = marketplace[:url].split(':')[2]
```

Comment:
@gnomet I need to clarify this a bit so you don't end up with wrong mental model. There's no preference or need to use only language's primitive types (strings, numbers, etc). The marketplace_url in this case could as well be passed as URI object from service. We just haven't had the need for anything other than string before. URI is a canonical way to represent urls in Ruby and it's a perfectly fine value type (not immutable though but neither are strings). Another example is monetary values represented as Money objects. Those are also value types and there's no problem whatsoever using them where it makes sense. In fact, the discovery api even puts (abstract versions of) URI and Money onto wire with Transit. Those are then parsed to language native presentations at receive time (e.g. JS in browser parses them as different wrapped objects than what Ruby does). What we don't want to do is to use objects for composite data types, i.e. to use a special purpose object to wrap a collection related properties. Hash (map) does much better job there.